BRANCHES.md

```diff
@@ -28,6 +28,15 @@ undocumented features, then you should be able to re-compile it without
 modification with any later release x.y'.z' with the same major version
 number, and your code will still build, be secure, and work.
 
+Note that new releases of Mbed TLS may extend the API. Here are some example of changes that are common in minor releases of Mbed TLS, and are not considered compatibility breaks:
```

Comment:
Possibly marginally clearer to say "... not considered API compatibility breaks" (to distinguish from ABI concerns).